demo/src/App.js

```diff
@@ -1,80 +1,50 @@
-
-const RedSlider = withStyles({
-  root: {
-    color: 'red'
-  }
-})(Slider);
-const BlueSlider = withStyles({
-  root: {
-    color: 'blue'
-  }
-})(Slider);
-const GreenSlider = withStyles({
-  root: {
-    color: 'green'
-  }
-})(Slider);
-const OrangeSlider = withStyles({
-  root: {
-    color: 'orange'
-  }
-})(Slider);
+import React, { PureComponent } from "react";
+import Slider from "@material-ui/core/Slider";
+import { withStyles } from "@material-ui/core/styles";
+import { MicroscopyViewer } from "../../src";
+import { source } from "./source-info";
+import "./App.css";
 
 export default class App extends PureComponent {
   constructor(props) {
     super(props);
-    this.handleRedSliderChange = this.handleRedSliderChange.bind(this);
-    this.handleGreenSliderChange = this.handleGreenSliderChange.bind(this);
-    this.handleBlueSliderChange = this.handleBlueSliderChange.bind(this);
-    this.handleOrangeSliderChange = this.handleOrangeSliderChange.bind(this);
     this.resize = this.resize.bind(this);
+    var sliderValues = {};
+    var colorValues = {};
+    var sliders = [];
+    const colorOptions = [
+      [255, 0, 0],
+      [0, 255, 0],
+      [0, 0, 255],
+      [255, 128, 0]
+    ];
+    Object.keys(source.channels).forEach((channel, i) => {
+      const sliderObj = {};
+      sliderValues[channel] = [0, 20000];
+      colorValues[channel] = colorOptions[i];
+      sliderObj[channel] = withStyles({
+        root: {
+          color: `rgb(${colorOptions[i]})`
+        }
+      })(Slider);
+      sliders.push(sliderObj);
+    });
     this.state = {
-      sliderValues: {
-        channel_0: [0, 20000],
-        channel_1: [0, 20000],
-        channel_2: [0, 20000],
-        channel_3: [0, 20000]
-      },
-      colorValues: {
-        channel_0: [255, 0, 0],
-        channel_1: [0, 255, 0],
-        channel_2: [255, 128, 0],
-        channel_3: [0, 0, 255]
-      },
+      sliderValues,
+      colorValues,
       viewHeight: window.innerHeight * 0.9,
       viewWidth: window.innerWidth * 0.7
     };
     this.max = 65535;
-    window.addEventListener('resize', this.resize);
+    this.sliders = sliders;
+    window.addEventListener("resize", this.resize);
   }
 
-  handleRedSliderChange(event, value) {
+  handleSliderChange(event, value, channel) {
+    var channelValue = {};
+    channelValue[channel] = value;
     this.setState({
-      sliderValues: { ...this.state.sliderValues, channel_0: value }
-    });
-  }
-
-  handleGreenSliderChange(event, value) {
-    this.setState({
-      sliderValues: { ...this.state.sliderValues, channel_1: value }
-    });
-  }
-
-  handleBlueSliderChange(event, value) {
-    this.setState({
-      sliderValues: { ...this.state.sliderValues, channel_3: value }
-    });
-  }
-
-  handleOrangeSliderChange(event, value) {
-    this.setState({
-      sliderValues: { ...this.state.sliderValues, channel_2: value }
+      sliderValues: { ...this.state.sliderValues, ...channelValue }
```

Comment:
Can you go ahead and change this to the call-back style? It's another eslint warning.